runtime/vm/resolvesupport.cpp

```diff
@@ -394,6 +394,15 @@ resolveClassRef(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA cpIndex, UDAT
 		}
 	}
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	if (J9_ARE_ANY_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_CHECK_VALUE_CLASS)) {
+		if (J9_ARE_NO_BITS_SET(accessModifiers, J9AccValueType)) {
+			setCurrentException(vmStruct, J9VMCONSTANTPOOL_JAVALANGINCOMPATIBLECLASSCHANGEERROR, NULL); // TODO: add detail message
```

Comment:
`TODO` comment is hidden; put it on a new line, and use `/* ... */` instead of `//`. According to the coding standards, `//` is not preferred for a comment.